docs/adr/adr-005-plugins.md

```diff
@@ -0,0 +1,273 @@
+# ADR 005: Plugins
+
+## Changelog
+
+- 2022-02-06: init commit
+
+## Context
+
+The modular design of celestia-node allows for the functionality it provides to be extended. The main blocker for creating custom light client applications is having a mechanism to access the internals of a celestia-node. Here we are suggesting that there be some interface that installs custom services into celestia-node, along with making the celestia-node cli more portable, so that light client developers can create their own binaries with the custom services that they create.
+
+## Alternative Approaches
+
+### Export most if not all the fields of a celestia-node object
+
+It should be possible to achieve the flexibility that a custom node requires, but it limits the UX for both developers and custom node users. For instance, running the custom node takes two binaries, both of which have to be installed using the associated versions. When creating a custom node, we have to implement redundant functionality, such as basic program structure, clis, and some tests.
+
+It also makes it more difficult for custom nodes to support different types of celestia nodes. With the plugin design, it is possible to run the same plugin on every node type without adding any code.
+
+Lastly, this approach wouldn't make it easy to run custom nodes with multiple modifications, whereas the plugin approach does.
+
+### Expose more services over rpc
+
+This would likely require a lot of overhead, and not eliminate any of the existing complexity.
```

Comment:
This is not an argument against the plugin approach but: rpc might be required either way though.